tests/PackageUrl.Tests.cs

```diff
@@ -115,5 +115,12 @@ public void TestJsonDeserialization(string data, string packageUrlStr)
                 Assert.NotNull(deserializedPurl.Qualifiers);
             }
         }
+        
+        [Theory]
+        [InlineData("\"Type\":\"npm\",\"Name\":\"foo\",\"Version\":\"1.2.3\"}")]
```

Comment:
There should probably also be variations here for if a malformed version or type is passed in, as well as an empty string.